From 340bb04ed8b9a455880b0cbac7228bb17a9679d8 Mon Sep 17 00:00:00 2001
Message-Id: <340bb04ed8b9a455880b0cbac7228bb17a9679d8@dist-git>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Tue, 7 Mar 2023 16:06:03 +0100
Subject: [PATCH] qemuxml2argvdata: Adjust maximum NUMA node used

We have couple of qemuxml2argvtest cases where up to 8 NUMA nodes
are assumed. These are used to check whether disjoint ranges of
host-nodes= is generated properly. Without prejudice to the
generality, we can rewrite corresponding XML files to use up to 4
NUMA nodes and still have disjoint ranges.

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Reviewed-by: Kristina Hanicova <khanicov@redhat.com>
Reviewed-by: Andrea Bolognani <abologna@redhat.com>
(cherry picked from commit d91ca262fba8c942449cb5f705f309fcf4baf05a)
Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=2185039
Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
---
 tests/qemuxml2argvdata/migrate-numa-unaligned.args           | 4 ++--
 tests/qemuxml2argvdata/migrate-numa-unaligned.xml            | 4 ++--
 tests/qemuxml2argvdata/numatune-memnode-restrictive-mode.xml | 4 ++--
 tests/qemuxml2argvdata/numatune-memnode.args                 | 4 ++--
 tests/qemuxml2argvdata/numatune-memnode.x86_64-5.2.0.args    | 4 ++--
 tests/qemuxml2argvdata/numatune-memnode.x86_64-latest.args   | 4 ++--
 tests/qemuxml2argvdata/numatune-memnode.xml                  | 4 ++--
 tests/qemuxml2xmloutdata/numatune-memnode.xml                | 4 ++--
 8 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/tests/qemuxml2argvdata/migrate-numa-unaligned.args b/tests/qemuxml2argvdata/migrate-numa-unaligned.args
index b50d93a12f..4786045358 100644
--- a/tests/qemuxml2argvdata/migrate-numa-unaligned.args
+++ b/tests/qemuxml2argvdata/migrate-numa-unaligned.args
@@ -17,9 +17,9 @@ XDG_CONFIG_HOME=/tmp/lib/domain--1-QEMUGuest/.config \
 -smp 32,sockets=32,cores=1,threads=1 \
 -object memory-backend-ram,id=ram-node0,size=20482048,host-nodes=3,policy=preferred \
 -numa node,nodeid=0,cpus=0,memdev=ram-node0 \
--object memory-backend-ram,id=ram-node1,size=675907584,host-nodes=0-7,policy=bind \
+-object memory-backend-ram,id=ram-node1,size=675907584,host-nodes=0-3,policy=bind \
 -numa node,nodeid=1,cpus=1-27,cpus=29,memdev=ram-node1 \
--object memory-backend-ram,id=ram-node2,size=24578457600,host-nodes=1-2,host-nodes=5,host-nodes=7,policy=bind \
+-object memory-backend-ram,id=ram-node2,size=24578457600,host-nodes=0,host-nodes=2,policy=bind \
 -numa node,nodeid=2,cpus=28,cpus=30-31,memdev=ram-node2 \
 -uuid 9f4b6512-e73a-4a25-93e8-5307802821ce \
 -display none \
diff --git a/tests/qemuxml2argvdata/migrate-numa-unaligned.xml b/tests/qemuxml2argvdata/migrate-numa-unaligned.xml
index e46b723acb..c060852297 100644
--- a/tests/qemuxml2argvdata/migrate-numa-unaligned.xml
+++ b/tests/qemuxml2argvdata/migrate-numa-unaligned.xml
@@ -6,8 +6,8 @@
   <vcpu placement='static'>32</vcpu>
   <numatune>
     <memnode cellid='0' mode='preferred' nodeset='3'/>
-    <memory mode='strict' nodeset='0-7'/>
-    <memnode cellid='2' mode='strict' nodeset='1-2,5-7,^6'/>
+    <memory mode='strict' nodeset='0-3'/>
+    <memnode cellid='2' mode='strict' nodeset='0-2,^1'/>
   </numatune>
   <os>
     <type arch='x86_64' machine='pc'>hvm</type>
diff --git a/tests/qemuxml2argvdata/numatune-memnode-restrictive-mode.xml b/tests/qemuxml2argvdata/numatune-memnode-restrictive-mode.xml
index 012c526460..2a640f5501 100644
--- a/tests/qemuxml2argvdata/numatune-memnode-restrictive-mode.xml
+++ b/tests/qemuxml2argvdata/numatune-memnode-restrictive-mode.xml
@@ -5,9 +5,9 @@
   <currentMemory unit='KiB'>24682468</currentMemory>
   <vcpu placement='static'>32</vcpu>
   <numatune>
-    <memory mode='restrictive' nodeset='0-7'/>
+    <memory mode='restrictive' nodeset='0-3'/>
     <memnode cellid='0' mode='restrictive' nodeset='3'/>
-    <memnode cellid='2' mode='restrictive' nodeset='1-2,5,7'/>
+    <memnode cellid='2' mode='restrictive' nodeset='1-2'/>
   </numatune>
   <os>
     <type arch='x86_64' machine='pc'>hvm</type>
diff --git a/tests/qemuxml2argvdata/numatune-memnode.args b/tests/qemuxml2argvdata/numatune-memnode.args
index 1564a0ddd6..dd0fea62e6 100644
--- a/tests/qemuxml2argvdata/numatune-memnode.args
+++ b/tests/qemuxml2argvdata/numatune-memnode.args
@@ -17,9 +17,9 @@ XDG_CONFIG_HOME=/tmp/lib/domain--1-QEMUGuest/.config \
 -smp 32,sockets=32,cores=1,threads=1 \
 -object memory-backend-ram,id=ram-node0,size=20971520,host-nodes=3,policy=preferred \
 -numa node,nodeid=0,cpus=0,memdev=ram-node0 \
--object memory-backend-ram,id=ram-node1,size=676331520,host-nodes=0-7,policy=bind \
+-object memory-backend-ram,id=ram-node1,size=676331520,host-nodes=0-3,policy=bind \
 -numa node,nodeid=1,cpus=1-27,cpus=29,memdev=ram-node1 \
--object memory-backend-ram,id=ram-node2,size=24578621440,host-nodes=1-2,host-nodes=5,host-nodes=7,policy=bind \
+-object memory-backend-ram,id=ram-node2,size=24578621440,host-nodes=0,host-nodes=2,policy=bind \
 -numa node,nodeid=2,cpus=28,cpus=30-31,memdev=ram-node2 \
 -uuid 9f4b6512-e73a-4a25-93e8-5307802821ce \
 -display none \
diff --git a/tests/qemuxml2argvdata/numatune-memnode.x86_64-5.2.0.args b/tests/qemuxml2argvdata/numatune-memnode.x86_64-5.2.0.args
index 81913e0e18..85f083efc9 100644
--- a/tests/qemuxml2argvdata/numatune-memnode.x86_64-5.2.0.args
+++ b/tests/qemuxml2argvdata/numatune-memnode.x86_64-5.2.0.args
@@ -18,9 +18,9 @@ XDG_CONFIG_HOME=/tmp/lib/domain--1-QEMUGuest/.config \
 -smp 32,sockets=32,cores=1,threads=1 \
 -object memory-backend-ram,id=ram-node0,size=20971520,host-nodes=3,policy=preferred \
 -numa node,nodeid=0,cpus=0,memdev=ram-node0 \
--object memory-backend-ram,id=ram-node1,size=676331520,host-nodes=0-7,policy=bind \
+-object memory-backend-ram,id=ram-node1,size=676331520,host-nodes=0-3,policy=bind \
 -numa node,nodeid=1,cpus=1-27,cpus=29,memdev=ram-node1 \
--object memory-backend-ram,id=ram-node2,size=24578621440,host-nodes=1-2,host-nodes=5,host-nodes=7,policy=bind \
+-object memory-backend-ram,id=ram-node2,size=24578621440,host-nodes=0,host-nodes=2,policy=bind \
 -numa node,nodeid=2,cpus=28,cpus=30-31,memdev=ram-node2 \
 -uuid 9f4b6512-e73a-4a25-93e8-5307802821ce \
 -display none \
diff --git a/tests/qemuxml2argvdata/numatune-memnode.x86_64-latest.args b/tests/qemuxml2argvdata/numatune-memnode.x86_64-latest.args
index 7cb7e659a4..6d4baebc83 100644
--- a/tests/qemuxml2argvdata/numatune-memnode.x86_64-latest.args
+++ b/tests/qemuxml2argvdata/numatune-memnode.x86_64-latest.args
@@ -18,9 +18,9 @@ XDG_CONFIG_HOME=/tmp/lib/domain--1-QEMUGuest/.config \
 -smp 32,sockets=32,cores=1,threads=1 \
 -object '{"qom-type":"memory-backend-ram","id":"ram-node0","size":20971520,"host-nodes":[3],"policy":"preferred"}' \
 -numa node,nodeid=0,cpus=0,memdev=ram-node0 \
--object '{"qom-type":"memory-backend-ram","id":"ram-node1","size":676331520,"host-nodes":[0,1,2,3,4,5,6,7],"policy":"bind"}' \
+-object '{"qom-type":"memory-backend-ram","id":"ram-node1","size":676331520,"host-nodes":[0,1,2,3],"policy":"bind"}' \
 -numa node,nodeid=1,cpus=1-27,cpus=29,memdev=ram-node1 \
--object '{"qom-type":"memory-backend-ram","id":"ram-node2","size":24578621440,"host-nodes":[1,2,5,7],"policy":"bind"}' \
+-object '{"qom-type":"memory-backend-ram","id":"ram-node2","size":24578621440,"host-nodes":[0,2],"policy":"bind"}' \
 -numa node,nodeid=2,cpus=28,cpus=30-31,memdev=ram-node2 \
 -uuid 9f4b6512-e73a-4a25-93e8-5307802821ce \
 -display none \
diff --git a/tests/qemuxml2argvdata/numatune-memnode.xml b/tests/qemuxml2argvdata/numatune-memnode.xml
index dd653c5d3b..9640eeb945 100644
--- a/tests/qemuxml2argvdata/numatune-memnode.xml
+++ b/tests/qemuxml2argvdata/numatune-memnode.xml
@@ -6,8 +6,8 @@
   <vcpu placement='static'>32</vcpu>
   <numatune>
     <memnode cellid='0' mode='preferred' nodeset='3'/>
-    <memory mode='strict' nodeset='0-7'/>
-    <memnode cellid='2' mode='strict' nodeset='1-2,5-7,^6'/>
+    <memory mode='strict' nodeset='0-3'/>
+    <memnode cellid='2' mode='strict' nodeset='0-2,^1'/>
   </numatune>
   <os>
     <type arch='x86_64' machine='pc'>hvm</type>
diff --git a/tests/qemuxml2xmloutdata/numatune-memnode.xml b/tests/qemuxml2xmloutdata/numatune-memnode.xml
index 104d2e6d4c..a117745bfb 100644
--- a/tests/qemuxml2xmloutdata/numatune-memnode.xml
+++ b/tests/qemuxml2xmloutdata/numatune-memnode.xml
@@ -5,9 +5,9 @@
   <currentMemory unit='KiB'>24682468</currentMemory>
   <vcpu placement='static'>32</vcpu>
   <numatune>
-    <memory mode='strict' nodeset='0-7'/>
+    <memory mode='strict' nodeset='0-3'/>
     <memnode cellid='0' mode='preferred' nodeset='3'/>
-    <memnode cellid='2' mode='strict' nodeset='1-2,5,7'/>
+    <memnode cellid='2' mode='strict' nodeset='0,2'/>
   </numatune>
   <os>
     <type arch='x86_64' machine='pc'>hvm</type>
-- 
2.40.0
